feature/dev-514/media-manager-integration

RamsesDijkstra opened the pull request:
Still testing wherether all the endpoints still work, here is the PR for the media-manager integration. Since its a big PR, might be worthwhile to go through it already...

RamsesDijkstra committed:
Processor WIP

RamsesDijkstra committed:
Added Media config

RamsesDijkstra committed:
Seems to compile...

RamsesDijkstra committed:
Add image EP

RamsesDijkstra committed:
Removed deprecated function calls

RamsesDijkstra committed:
WIP

RamsesDijkstra committed:
Rest of endpoints

RamsesDijkstra committed:
Get EPs WIP

RamsesDijkstra committed:
Video Get EP

RamsesDijkstra committed:
Audio Get EP

RamsesDijkstra committed:
Chi -> Gin param validation

RamsesDijkstra committed:
Get Asset EP

RamsesDijkstra committed:
Small compilation fix

RamsesDijkstra committed:
PR fix

RamsesDijkstra committed:
More fixes..

RamsesDijkstra committed:
...

jor-rit commented on config/media.go:
```diff
@@ -0,0 +1,25 @@
+	Assetpath string `yaml:"asset_path" envconfig:"RENDER_ASSET_PATH"`
+	LogLevel  int8   `yaml:"loglevel"  envconfig:"RENDER_LOGLEVEL"`
+	// Enabled pprof http endpoints. If enabled, point your browser at /debug/pprof
+	PProfAPI bool `yaml:"pprof_api" envconfig:"RENDER_PPROF_API"`
```
this one can be dropped now, controller already has its own pprof api endpoints

jor-rit commented on config/media.go:
```diff
@@ -0,0 +1,25 @@
+package config
+
+type Media struct {
+	Address   string `yaml:"bind_address" envconfig:"RENDER_BIND_ADDRESS"`
```
address & port can also be dropped, UC already has its own

jor-rit commented on pkg/media/media.go:
```diff
@@ -0,0 +1,242 @@
+	processor *processor.Processor
+}
+
+var Tsizes = map[string]int{
```
hmm, now see the same map in images.go. Should probably just (re)use that one?

RamsesDijkstra committed:
Asset3d upload / config changes

RamsesDijkstra committed:
Updated RenderAuto

RamsesDijkstra committed:
API config

RamsesDijkstra committed:
Merge branch 'develop' into feature/DEV-514/media-manager-integration

RamsesDijkstra committed:
CI fix

RamsesDijkstra committed:
...

RamsesDijkstra committed:
Updated seeder

RamsesDijkstra committed:
VideoGet fix, added fonts

RamsesDijkstra committed:
Removed unused env param

RamsesDijkstra committed:
Deltrack EP, moved types

RamsesDijkstra committed:
CI fix

jor-rit committed:
chore: move fonts outside of storage directory

To make deployment a bit easier.

The storage is remote/shared storage on actual deployments. This allows
the whole directory can be mounted in 1 go.

The fonts are static files needed at runtime, can be included in the
docker image.

jor-rit committed:
build: add fonts to docker image

RamsesDijkstra committed:
Error handling on GET EPs

RamsesDijkstra committed:
Merge branch 'develop' into feature/DEV-514/media-manager-integration
